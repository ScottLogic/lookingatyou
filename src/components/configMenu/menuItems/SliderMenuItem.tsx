import { Slider, Tooltip } from '@material-ui/core';
import React from 'react';
import { PartialConfig } from '../../../store/actions/config/types';
import { HelpWith } from '../Help';

function ValueLabelComponent(props: any) {
    const { children, open, value } = props;
    const popperRef: any = React.useRef(null);

    React.useEffect(() => {
        if (popperRef.current) {
            popperRef.current!.update();
        }
    });

    return (
        <Tooltip
            PopperProps={{
                popperRef,
            }}
            open={open}
            enterTouchDelay={0}
            placement="top"
            title={value}
        >
            {children}
        </Tooltip>
    );
}

export interface ISliderMenuItemProps {
    name: string;
    configName: string;
    step: number;
    min?: number;
    max?: number;
    onValidInput: (payload: PartialConfig) => void;
    defaultValue: number;
    helpWith: HelpWith;
}

const SliderMenuItem = React.memo(
    (props: ISliderMenuItemProps) => {
        function onChange(ignore: object, value: number | number[]) {
            props.onValidInput({
                [props.configName]: value,
            });
        }

        return (
            <div data-tip={true} data-for={HelpWith[props.helpWith]}>
                <label>{props.name}</label>
                <Slider
                    ValueLabelComponent={ValueLabelComponent}
                    aria-label="custom thumb label"
                    defaultValue={props.defaultValue}
                    value={props.defaultValue}
                    onChange={onChange}
                    min={props.min}
                    max={props.max}
                    step={props.step}
                    className="slider"
                />
            </div>
        );
    },
    (previous, next) => {
        return (
            previous.name === next.name &&
            previous.defaultValue === next.defaultValue
        );
    },
);
export default SliderMenuItem;
