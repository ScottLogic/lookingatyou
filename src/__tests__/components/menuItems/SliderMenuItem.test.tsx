import { shallow } from 'enzyme';
import React from 'react';
import SliderMenuItem, {
    ISliderMenuItemProps,
} from '../../../components/configMenu/menuItems/SliderMenuItem';

let props: ISliderMenuItemProps;

it('should render correctly', () => {
    const wrapper = shallow(<SliderMenuItem {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
});
