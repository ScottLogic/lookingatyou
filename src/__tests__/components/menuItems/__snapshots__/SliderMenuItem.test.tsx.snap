// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
"<div data-tip={true} data-for={[undefined]}>
  <label>
    test
  </label>
  <WithStyles(ForwardRef(Slider)) ValueLabelComponent={[Function: ValueLabelComponent]} aria-label=\\"custom thumb label\\" defaultValue={30} value={30} onChange={[Function: onChange]} min={1} max={[undefined]} step={1} className=\\"slider\\" />
</div>"
`;
